### PR TITLE
feat: add JWT support

### PR DESCRIPTION
**Pull request by bra-i-am**
## Description

This PR allows you to make requests using a JWT

## How to test

1. Install the plugin with these changes
2. Install the Forum Tutor Plugin with `tutor plugins install forum` && `tutor plugins enable forum`
3. Create a new application from `http://local.edly.io:8000/admin/oauth2_provider/application/`. Associate a user with sufficient permissions and add to the **Redirect uris** field: http://local.edly.io:8000/
4. Add the following setting to the tenant you are using:
``` JSON
"EOX_TAGGING_DEFINITIONS": [
       {
           "access": "public",
           "owner_object": "site",
           "tag_type": "subscription_level",
           "validate_target_object": "user"
       },
       {
           "access": "public",
           "owner_object": "site",
           "tag_type": "subscription_level_enrollment",
           "validate_target_object": "courseenrollment"
       },
       {
           "access": "public",
           "owner_object": "site",
           "tag_type": "subscription_level_course",
           "validate_target_object": "OpaqueKeyProxyModel"
       },
       {
           "access": "public",
           "owner_object": "user",
           "tag_type": "subscription_level_user",
           "validate_target_object": "user"
       }
   ]

```
> [!NOTE]
> For more information about the eox-tagging configuration please follow the [Test Cases Doc](https://docs.google.com/document/d/1VnjGF6Ri1Bvgbcd8AulM5jt68U7BmYJLVV2H5Bi5VOg/edit#heading=h.9sztqeti556d)

5. Open `Postman` and import the Collection and Environment at `docs/resources` adding the required env info according to the previously created application 
6. Generate a JWT using the `Authorization` POST route from the collection; the generated token will be added automatically to the environment
7. Run the endpoints from the collection previously imported following the ascending numeration

[JIRA ISSUE AP-1111](https://edunext.atlassian.net/browse/AP-1111)

### CODE REVIEW

**Review by magajh — COMMENTED** [review]
Thank you @bra-i-am!

File: docs/resources/eox-tagging.postman_collection.json [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+				"method": "DELETE",
+				"header": [
+					{
+						"key": "Authorization",
+						"value": "JWT {{token}}",
+						"type": "text"
+					}
+				],
+				"url": {
+					"raw": "{{domain}}/eox-tagging/api/v1/tags/{{user_tag_key}}",
```
Comment (BryanttV) [inline docs/resources/eox-tagging.postman_collection.json:370]: Without the slash it doesn't work
```suggestion
					"raw": "{{domain}}/eox-tagging/api/v1/tags/{{user_tag_key}}/",
```

**Review by BryanttV — COMMENTED** [review]
Hi @bra-i-am, I tested all the endpoints locally in Redwood, and it allows using JWT, thanks!. I just left a comment about the URL in the DELETE methods in the Postman collection.

**Review by BryanttV — COMMENTED** [review]
@bra-i-am, Sorry for not noticing it before, but I think we should make a bump version.

File: eox_tagging/api/v1/viewset.py [hunk 2]
```diff
@@ -207,7 +208,7 @@ class TagViewSet(viewsets.ModelViewSet):
     """Viewset for listing and creating Tags."""
 
     serializer_class = TagSerializer
-    authentication_classes = (BearerAuthentication, SessionAuthentication)
+    authentication_classes = (BearerAuthentication, SessionAuthentication, JwtAuthentication,)
```
Comment (magajh) [inline eox_tagging/api/v1/viewset.py:211]: @bra-i-am unless we have a good reason to enforce immutability for the `authentication_classes` here, I think it's best to stick with lists as we did in the other plugins: 
https://github.com/eduNEXT/eox-manage/pull/25/files#diff-80eed3c6af56abe3c9be25ddfededbc269a567eaa8435f015f5104dc6bdaec1dR46
https://github.com/eduNEXT/eox-support/pull/17/files#diff-c21fbddde5a151dabbde2f7f31d52fe9949c97871c7ca36189918a6c2e0004f8R85

**Review by magajh — APPROVED** [review]

**Review by BryanttV — APPROVED** [review]
LGTM! thanks

**Pull request merged**